### ISSUE TITLE
Why Orientation in Listener Object and Source Object are storage differently ?

### ISSUE THREAD

**Issue by Pmea** [google/songbird#8]
I looked into the code of Listener Object and Source Object, I noticed the Orientation is not stored in the same way. For Source, it's stored in two 1D-arrays (named _forward an _up), whereas for Listener, it's stored in one 1D-array (named _tempMatrix4).
I'm not sure to understand why it's programmed differently ? Is it for a specific usage of Listener ?

**Comment by drewbitllama** [comment]
Hi Pierre,

It's due to listener requiring a rotation matrix in order to rotate the
incoming soundfield where as we need forward and up vector formats to
easily compute directivity of the source (which is independent of the
listener's orientation )

Hope that helps make sense.

Cheers,
Drew


On Oct 19, 2017 6:46 AM, "Pierre Mahé" <notifications@github.com> wrote:

I looked into the code of Listener Object and Source Object, I noticed the
Orientation is not stored in the same way. For Source, it's stored in two
1D-arrays (named _forward an _up), whereas for Listener, it's stored in one
1D-array (named _tempMatrix4).
I'm not sure to understand why it's programmed differently ? Is it for a
specific usage of Listener ?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/google/songbird/issues/8>, or mute the thread
